### PR TITLE
submit week5-6 exercises

### PR DESCRIPTION
**Pull request by kerenor557100**
Hi Itay,
please review my pull request consisting my submission of week5-6 (Yam Mesika) exercises.
thank you!
Keren Or

### CODE REVIEW

File: zo_haderech.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#submitted by Keren Or Hadad 208025205
+
+from os import listdir as dirs
+
+# part 5.1
+def zo_haderech(path: str):
```
Comment (itayday) [inline zo_haderech.py:6]: Please add documantation to your function Keren :)

File: zo_haderech.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    item = []
+    for filename in dirs(path):
```
Comment (itayday) [inline zo_haderech.py]: Are there ways to simplify your code? (For example with list comprehension, or with string methods (try to read about "startswith")

File: zo_haderech.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#submitted by Keren Or Hadad 208025205
+
+from os import listdir as dirs
+
+# part 5.1
+def zo_haderech(path: str):
+    item = []
```
Comment (itayday) [inline zo_haderech.py]: Please use a different name for that list (one that will be more informative about what it does).

File: en_li_vinigret.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#submitted by Keren Or Hadad 208025205
+import datetime
+import random
+
+
+def en_li_vinigret(start_date: datetime, end_date: datetime):
```
Comment (itayday) [inline en_li_vinigret.py:6]: Please add documantation to your function.

File: interleave.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#submitted by Keren Or Hadad 208025205
+# part 5.4
+import itertools
+
+
+def interleave(*lists):
+    if len(set(len(_) for _ in lists)) > 1:
+        raise ValueError("Lists are not all the same length!")
+    joint = list(itertools.chain(*lists))
+    for l_idx, li in enumerate(lists):
```
Comment (itayday) [inline interleave.py]: The names of the variables here should be more informative :)

File: interleave.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        joint[l_idx::len(lists)] = li
+    return joint
+
+print(interleave('abc', [1, 2, 3], ('!', '@', '#')))
```
Comment (itayday) [inline interleave.py:21]: Nice work :)

File: cup_of_join.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#submitted by Keren Or Hadad 208025205
+
+#part 5.2
+
+def cup_of_join(*args, sep = '-'):
```
Comment (itayday) [inline cup_of_join.py:5]: Please add documantation.

File: cup_of_join.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    lst_return = []
+    for arg in args:
```
Comment (itayday) [inline cup_of_join.py]: Can we simplify this loop by using list comprehension?

File: get_recipe_price.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
+#submitted by Keren Or Hadad 208025205
+
+
+def get_recipe_price(d_ing: dict,  *args, optionals = []):
```
Comment (itayday) [inline get_recipe_price.py]: Add documantation, and please use more informative names for your parameters.
(Notice that there were specific names that you have been requested to use).

File: zo_haderech.py (hunk 1 above)
Comment (kerenor557100) [reply zo_haderech.py:6]: done

File: zo_haderech.py (hunk 2 above)
Comment (kerenor557100) [reply zo_haderech.py]: done

File: zo_haderech.py (hunk 3 above)
Comment (kerenor557100) [reply zo_haderech.py]: done

File: en_li_vinigret.py (hunk 4 above)
Comment (kerenor557100) [reply en_li_vinigret.py:6]: done

File: interleave.py (hunk 5 above)
Comment (kerenor557100) [reply interleave.py]: done

File: cup_of_join.py (hunk 7 above)
Comment (kerenor557100) [reply cup_of_join.py:5]: done

File: cup_of_join.py (hunk 8 above)
Comment (kerenor557100) [reply cup_of_join.py]: done

File: get_recipe_price.py (hunk 9 above)
Comment (kerenor557100) [reply get_recipe_price.py]: done

File: interleave.py (hunk 6 above)
Comment (kerenor557100) [reply interleave.py:21]: thank you

**Pull request merged**